searx/searx
# Use Wikipedia's REST v1 API

## What does this PR do?

After the changes in #2102 I realised that sometimes the Spanish Wikipedia still returns disambiguation paragraphs in the infobox. The new REST API doesn't have that problem.

This also fixes a little issue with the infobox toggle button when sometimes the button is hidden even though there is more content to be shown.

## Why is this change important?

This is the API that Wikipedia currently uses to retrieve an article's preview when you hover your mouse over a link in Wikipedia. Therefore, using this API is way simpler to use and should also be more accurate than the older, general purpose API that's not really meant for article previews.

## How to test this PR locally?

* Try to query an article with a disambiguation paragraph, such as `:es !wp naruto`. No disambiguation (*desambiguación*) paragraph should be shown.
* Same tests as #2102.
* The toggle issue could be harder to reproduce because the error happens when the image is loaded a fraction of a second after the JS code was run. But try `:en !wp Port Dover` with master and see if the toggle button is hidden even though there's a hidden image. That bug should never happen in this branch.

## Author's checklist

The official documentation (https://en.wikipedia.org/api/rest_v1/) mentions two rules:
> 
> * Limit your clients to no more than 200 requests/s to this API. Each API endpoint's documentation may detail more specific usage limits.
> * Set a unique User-Agent or Api-User-Agent header that allows us to contact you quickly. Email addresses or URLs of contact pages work well.

I set the Wikipedia engine to send searx's user agent to comply with that.
